src/model/attributes/relations/MorphTo.ts

```diff
@@ -0,0 +1,228 @@
+import { Schema as NormalizrSchema } from 'normalizr'
+import { assert } from '../../../support/Utils'
+import { Schema } from '../../../schema/Schema'
+import { Element, Collection } from '../../../data/Data'
+import { Query } from '../../../query/Query'
+import { Database } from '../../../database/Database'
+import { Model } from '../../Model'
+import { NonEnumerable } from '../../decorators/NonEnumerable'
+import { Relation } from './Relation'
+import { schema as Normalizr } from 'normalizr'
+
+export class MorphTo extends Relation {
+  /**
+   * The database instance.
+   */
+  @NonEnumerable
+  private _database!: Database
+
+  /**
+   * The field contains all related models.
+   */
+  private _relatedModels: Model[]
+
+  /**
+   * The field contains all the related models.
+   */
+  private _relatedSchemas: Record<string, Normalizr.Entity> = {}
+
+  /**
+   * The field contains all related types.
+   */
+  private _relatedTypes: Record<string, Model> = {}
+
+  /**
+   * The field contains all related types.
+   */
+  private _relatedQueries: Record<string, Query> = {}
+
+  /**
+   * The field name that contains id of the parent model.
+   */
+  protected morphId: string
+
+  /**
+   * The field name that contains type of the parent model.
+   */
+  protected morphType: string
+
+  /**
+   * The associated key of the child model.
+   */
+  protected ownerKey: string
+
+  /**
+   * Create a new morph-to relation instance.
+   */
+  constructor(
+    parent: Model,
+    _relatedModels: Model[],
+    morphId: string,
+    morphType: string,
+    ownerKey: string
+  ) {
+    super(parent, parent)
+    this._relatedModels = _relatedModels
+    this.morphId = morphId
+    this.morphType = morphType
+    this.ownerKey = ownerKey
+  }
+
+  /**
+   * Get all related models for the relationship.
+   */
+  getRelateds(): Model[] {
+    return this._relatedModels
+  }
+
+  /**
+   * Define the normalizr schema for the relation.
+   */
+  define(schema: Schema): NormalizrSchema {
+    return schema.union(this._relatedSchemas, (_value, parentValue, _key) => {
+      // HACK: Assign missing parent id since the child model is not related back and `attach` will not be called
+      const type: string = parentValue[this.morphType]
+      const model: Model = this.$getRelatedModel(type)
+      const key: string = this.ownerKey || (model.$getKeyName() as string)
+      parentValue[this.morphId] = _value[key as string]
```

Comment:
Attaching the parent value to the child in the union schema callback since attach will not be called when the parent and child are both set to the same model.